### ISSUE TITLE
Build failure on recent ubuntu/debian

### ISSUE THREAD

**Issue by avalentino** [PyTables/PyTables#442]
```
$ python setup.py build
* Using Python 2.7.9 (default, Apr  2 2015, 15:33:21) 
/usr/bin/ld: impossibile trovare -lhdf5
collect2: error: ld returned 1 exit status
.. ERROR:: Could not find a local HDF5 installation.
   You may need to explicitly state where your local HDF5 headers and
   library can be found by setting the ``HDF5_DIR`` environment
   variable or by using the ``--hdf5`` command-line option.
```

The problem is related to the new layout of the debian hdf5 package that installs libraries and headers in a hdf5/serial sub-directory:

```
/usr/include/hdf5/serial/hdf5.h
/usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so
```

I'm working on a solution based on pkg-config and I'm planning to setup  PR in a couple of days.
Alternative fixes are welcome.

**Comment by andreabedini** [comment]
I'm surprised by the new directory layout. +1 to a solution based on pkg-config!

**Comment by avalentino** [comment]
> @avalentino on what versions of ubuntu and debian does this happen?

Ubuntu 15.04 and Debian 8 (Jessie), both have been released in April 2015.

**Comment by FrancescAlted** [comment]
Oh, that's a nice time to fix these platforms indeed.  I am +1 on pkg-config too.  Also, it would be nice to have a look at what h5py is doing for solving this.

**Comment by avalentino** [comment]
AFAIKT h5py uses the python-pkgconfig package but I would like to avoid to add new pre-requisites.

My plan is to use pkg-config only if it is available and IMO it is more probable to find the command line pkg-config tool already installed on a system rather then python-pkgconfig.

**Comment by andreabedini** [comment]
I agree that including another "setup time" dependency would be unfortunate. I really don't understand the logic of this change though. On debian/ubuntu, I can't install just hdf5 and then compile a simple file that `#include "hdf5.h"` without specifying some include path? Do I understand right? That to me is weird, so it assumes that everybody uses some more complex build system (autoconf/cmake).

**Comment by avalentino** [comment]
> On debian/ubuntu, I can't install just hdf5 and then compile a simple file that #include "hdf5.h" without specifying some include path?

yes, you have to specify paths for headers and libs or use build helpers like h5cc and h5c++.

**Comment by FrancescAlted** [comment]
Fixed by merging PR #444

**Comment by SabineHaas** [comment]
I'd like to reopen the issue because I get the same error again when installing PyTables with python 3.7. ...[see failing build here!](https://travis-ci.org/wind-python/windpowerlib/jobs/496488756) I do not have this problem with python 3.5 or 3.5.

Is there any change that this will get fixed for python 3.7 soon?

**Comment by FrancescAlted** [comment]
No that I am aware of.  We would appreciate some input on how to fix this in TravisCI (for example, I am not having problems with Python 3.7.1 when compiling via the usual `python setup.py build_ext -i` on my Mac box, but it might be because Anaconda has a better handle of the paths for hdf5).
